File: Sources/VideoEditCore/Segment.swift
import Foundation

public struct Segment: Sendable, Codable, Identifiable {
    public let id: UUID
    public var start: Double
    public var end: Double
    public var text: String
    public var words: [WordTimestamp]

    public init(start: Double, end: Double, text: String, words: [WordTimestamp] = []) {
        self.id = UUID()
        self.start = start
        self.end = end
        self.text = text
        self.words = words
    }
}

public struct WordTimestamp: Sendable, Codable {
    public let word: String
    public let start: Double
    public let end: Double

    public init(word: String, start: Double, end: Double) {
        self.word = word
        self.start = start
        self.end = end
    }
}

File: crates/videoforge-core/src/segment.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct WordTimestamp {
    pub word: String,
    pub start: f64,
    pub end: f64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Segment {
    pub id: u64,
    pub start: f64,
    pub end: f64,
    pub text: String,
    pub words: Vec<WordTimestamp>,
}

impl Segment {
    pub fn new(start: f64, end: f64, text: impl Into<String>, words: Vec<WordTimestamp>) -> Self {
        Self {
            id: 0,
            start,
            end,
            text: text.into(),
            words,
        }
    }
}

impl WordTimestamp {
    pub fn new(word: impl Into<String>, start: f64, end: f64) -> Self {
        Self { word: word.into(), start, end }
    }
}
